src/main/java/com/couchbase/lite/Database.java

```diff
@@ -1187,6 +1214,28 @@ public synchronized boolean open() {
             optimizeSQLIndexes(); // runs ANALYZE query
         }
 
+        // NOTE: Migrate attachment directory path if necessary
+        // https://github.com/couchbase/couchbase-lite-java-core/issues/604
+        File obsoletedAttachmentStorePath = new File(getObsoletedAttachmentStorePath());
+        if (obsoletedAttachmentStorePath != null && obsoletedAttachmentStorePath.exists() && obsoletedAttachmentStorePath.isDirectory()) {
+            File attachmentStorePath = new File(getAttachmentStorePath());
+            if (attachmentStorePath != null && !attachmentStorePath.exists()) {
+                boolean success = obsoletedAttachmentStorePath.renameTo(attachmentStorePath);
+                if (!success) {
+                    Log.e(Database.TAG, "Could not rename attachment store path");
+                    database.close();
+                    return false;
+                }
+            }
+
+        }
+        // NOTE: obsoleted directory is /files/<database name>/attachments/xxxx
+        //       Needs to delete /files/<database name>/ too
+        File obsoletedAttachmentStoreParentPath = new File(getObsoletedAttachmentStoreParentPath());
+        if (obsoletedAttachmentStoreParentPath!=null&&obsoletedAttachmentStoreParentPath.exists()){
+                obsoletedAttachmentStoreParentPath.delete();
+        }
+
```

Comment:
Line 1234-1237 should be out side of if block (line 1220 - 1231).

Previous version of CBL Android/Java creates the directory, `/files/<database name>/` in case of no attachment. But does not create `/files/<database name>/attachments`.